drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -328,12 +373,24 @@ local function set_setpoint(setpoint)
       {feature_bitmap = clusters.Thermostat.types.ThermostatFeature.AUTOMODE}
     ) > 0
 
+    local cooling_setpoint_range = device:get_latest_state(
```

Comment:
What is the reason for using 0 and 100 as the default min and max? Also, what are the expected units? I think there are a few issues with unit ambiguity that we will need to address

Reply:
@nickolas-deboom the `get_latest_state()` function returns two values as described [here](https://developer.smartthings.com/docs/edge-device-drivers/device.html?highlight=get_latest#st.Device.get_latest_state). The first return value is the state.value present for the attribute, the second return is the state. Does the second return value include the units? You can look at the examples a few lines above this to see how this might work by getting the unit from the second return value and then doing the conversion as needed: 
```
    local cached_heating_val, heating_setpoint = device:get_latest_state(
      cmd.component, capabilities.thermostatHeatingSetpoint.ID,
      capabilities.thermostatHeatingSetpoint.heatingSetpoint.NAME,
      0, { value = 0, unit = "C" }
    )
    if heating_setpoint and heating_setpoint.unit == "F" then
      cached_heating_val = utils.f_to_c(cached_heating_val)
    end
```

Reply:
The second return value indeed includes the units. I made this update in 7d6d886571b626d615efd8660e077053ea45e6ff.